api/lib/database/methods.js

```diff
@@ -95,6 +95,131 @@ const getNotifications = ({ hasura_url, hasura_admin_key }) => (
   )
 }
 
+const getNetwork = ({ hasura_url, hasura_admin_key }) => () => async (id) => {
+  const storedNetwork = await read({
```

Comment:
you can run all 3 reads in parallel. but better use `query` and specify all 3 requests in 1 query.